### DISCUSSION TITLE
Deployment Status - Why doesn't it show appropriate status in Mia Console?

### DISCUSSION THREAD

**Discussion by phil-hotchkiss** [mia-platform/community#387] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

v12.0.2

### Description

I have two related issues going on. My company is using the pull based (ArgoCD) deployment model. When we hit the Deploy button, we are left with a seemingly infinite spinner. In using my browser tools, I was able to determine that the UI is polling https://{mia-console}/api/deploy/projects/65a80a9b9c1f85310293ea6d/pipelines/1314558823/status/?environment={env}. The response is returning as `pending` and thus the UI continues to wait for it to be successful. After some deeper diving, I went to the /deploy/history page and noticed that these deployments never seem to be resolved as `success`, so I determined I needed to add a github action to update the deployment status because the network tab was showing https://github.com/Mortgage-Research-Center/architectmiamigrationtesting-configurations/commit/{commitId}/checks as the URL that was being leveraged to determine the status. I expected that once I added the update to the deployment status and marked it successful, I'd be able to see that represented within Mia Console, however, it continues to show as `pending`. What else needs to be updated to ensure Mia Console knows the deployment was successful? 

![Screenshot 2024-02-06 at 4 29 52 PM](https://github.com/mia-platform/community/assets/136851567/b2a71138-b9ef-4318-b5aa-e60b05053116)

![Screenshot 2024-02-06 at 4 33 00 PM](https://github.com/mia-platform/community/assets/136851567/4e5aabc2-c8af-431e-bbd6-2999e106b507)

### Actual Outcome

Mia Console shows a Pending status of a deployment

### Expected Outcome

Mia Console shows a success status of a deployment when the deployment in Github is marked successful.

**Comment by davidebianchi** [comment]
Hi @phil-hotchkiss! Thanks for the information, we will investigate about it. Do you have some pipelines providers configured in your project, is it correct? 
For the pull based deployment, you could remove this provider and it should work correctly the deploy, without the pending status and updating the deploy history.

**Reply by phil-hotchkiss** [reply]
For context, we are using an on-premises deployment of the Console. I do not have a provider configured for CI/CD. I was originally not doing any Github actions as it relates to the deployment, but the deployments were never showing as completed. That was how I started down this path of updating the deployment status.  Are you referencing the attached image for determining if we have pipeline providers configured for the project?  

![image](https://github.com/mia-platform/community/assets/136851567/bba9f6e8-ef4d-4fc1-a897-1fb9098e931e)

**Reply by davidebianchi** [reply]
Thank you for the information. From the image you attached, in the github provider has the `CI/CD Tool` capability. You can also disable it, so that in project creation is not possible to select.
Can you check inside the providers page in the company settings? [Here docs reference](https://docs.mia-platform.eu/docs/console/project-configuration/project-settings#providers)

**Reply by phil-hotchkiss** [reply]
The previously attached image is from the Company page. I went into the provider and unchecked CI/CD tool as an option for the provider, so now that CI/CD Tool pink box is no longer there. I executed the deployment steps and the same behavior was exhibited. Let me know if there is anything else I should explore. Thanks!

**Reply by davidebianchi** [reply]
Oh, sorry, I made a mistake in writing. I meant in the project settings section in the provider tab

**Reply by phil-hotchkiss** [reply]
I do not see a way to remove an existing CI/CD provider from an already existing project. It seems that a CI/CD provider is required in order for the Deploy screen to work though, right? If one is not set, I get the following screen: 

![image](https://github.com/mia-platform/community/assets/136851567/184953d6-729b-496c-81a0-d371330cdb1a)

Our understanding is that the hook that commits the helm values that were returned from our extension will be committed to the repository when the Deploy button is hit. 

As an additional note, the deployment strategy is currently set to `Push` and says it's missing a CI/CD provider. If I switch it to `Pull`, I still have the same screen outlined above. 

This seems true in any configuration that if the deployment strategy is set to pull, it says it's missing a CI/CD provider. Is there maybe a bug related to the pull based deployment model being enabled? 

Let me know if I'm missing something! Thanks!

**Reply by davidebianchi** [reply]
Ok, thank you for the detail. We will try with this settings and verify if there is a bug related. Probably the ci/cd should be set also if unused (as in the pull mode).

> Our understanding is that the hook that commits the helm values that were returned from our extension will be committed to the repository when the Deploy button is hit.

Yes, this is correct.

Those configurations are on a newly created project, right?
In the first project, is the strategy set to `push` or `pull`? If it is set to push, can you try to set to `pull` mode and retry the deploy (without removing the ci/cd provider)? It should remove the infinite spinner.

**Reply by phil-hotchkiss** [reply]
Any circumstance in which `pull` is selected seems to produce the screen above.

**Reply by phil-hotchkiss** [reply]
I think I found the bug. If the screen is refreshed after changing to `pull`, you get the screen above. The `pull` based deployment seems to work if you first navigate to the deploy screen while in `push` mode, go back to settings, set deployment strategy to `push`, click the deploy tab again and then deploy. But again, if you refresh your screen or log out and then back in, the CI/CD tool not connected screen shows up.

**Reply by davidebianchi** [reply]
Thank you for the info, we replicated and resolve it. It will be released in a few moments with the v12.1.2 of the Console

**Reply by fredmaggiowski** [reply, marked as answer]
Hi @phil-hotchkiss we've just released Console v12.1.2 ([release note here](https://docs.mia-platform.eu/docs/release-notes/v12.1.2)) with the fix, you can install it using the Console helm chart v11.1.2.

If you have any further issues don't hesitate to let us know or to contact your Mia-Platform referent!

**Reply by phil-hotchkiss** [reply]
Thanks a bunch! I have been able to use the deploy tab as expected after switching the Deployment Strategy to `pull`. Much appreciated for the quick turnaround!
